addon/-private/router-dsl-ext.js

```diff
@@ -44,8 +44,6 @@ EmberRouterDSL.prototype.mount = function(_name, _options) {
   let localFullName = 'application';
   let routeInfo = assign({ localFullName }, engineInfo);
 
-  this.options.addRouteForEngine(fullName, routeInfo);
```

Comment:
On further inspection, this is needed. Notice that `routeInfo` contains `localFullName` which is `"application"`. Which is different from the `localFullName` used below.